bin/cron/cleanup_workshop_attendance_codes

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/only_one'
+
+def main
+  ten_days_ago = Time.now - 10.days
+  three_days_ago = Time.now - 3.days
+  Pd::Session.
+    where('end > ?', ten_days_ago).
+    where('end < ?', three_days_ago).
+    find_each(&:remove_code)
+end
```

Comment:
Why between 3 and 10 days? Note we may be reinspecting sessions that have already given up codes, and we may miss older ones if we skip a week for some reason. This should be idempotent and complete. This should include a filter for `.where.not(code: nil)` and it should NOT include an upper bound (> 10 days ago). We should be able to run this any time and it should clean up everything it needs to and no more.

The logic here should be in the model, ideally with unit tests. I'd like this cronjob to simply call something like `Pd::Session.cleanup_old_codes` which will contain all the logic.

Do we care if the workshop has ended, or only the scheduled session start and end days?

Reply:
Conceptually, the code cleanup logic doesn't need to be aware of how often it's run, so this 10 day cap is a red flag. The frequency is just a scheduling thing. Whenever it is run, it should clean up what needs to be cleaned.